src/renderer/redux/reducers/api.ts

```diff
@@ -60,18 +72,36 @@ export function apiReducer(
                     [requestId]: requestState,
                 };
             }
+
+            // if an error happened in request
+            // return the state whithout updated LAST_API_SUCCESS_ID
             if (action.error) {
                 return {
                     ...returnState,
                 };
             }
+
+            // autherwise format lastApiSuccess for the refresh feature
+            const lastApiSuccess = {
+                ...state[LAST_API_SUCCESS_ID],
+            };
+            lastApiSuccess.lastSuccess = lastApiSuccess.data;
+            lastApiSuccess.lastTime = lastApiSuccess.data?.time || 0;
+            lastApiSuccess.data = returnState[requestId].data;
+
+            // no error is happened
+            // return all the API state and the lastApiSuccess dedicated channel
             return {
                 ...returnState,
-                [LAST_API_SUCCESS_ID]: returnState[requestId],
+                [LAST_API_SUCCESS_ID]: lastApiSuccess,
```

Comment:
I see, this is the most relevant code/logic change in this PR (the rest is comments and formatting, which are also useful).
I do not understand all the details, but I approve the PR.